packages/docusaurus-plugin-content-pages/src/index.ts

```diff
@@ -66,15 +71,22 @@ export default function pluginContentPages(
         return;
       }
 
-      const {addRoute} = actions;
+      const {addRoute, createData} = actions;
 
+      const configPath = await createData(
+        `${docuHash('config')}.json`,
+        JSON.stringify({siteConfig: context.siteConfig}, null, 2),
+      );
       await Promise.all(
         content.map(async (metadataItem) => {
           const {permalink, source} = metadataItem;
           addRoute({
             path: permalink,
             component: source,
             exact: true,
+            modules: {
+              config: configPath,
+            },
```

Comment:
maybe the solution is just something like

```
modules: {
  config: "@generated/docusaurus.config",
},
```

never tried this yet but that's worth giving it a shot ;)

Reply:
I changed to use generated file

Reply:
My first implementation was using `docusaurus.config` but on second thoughts using a core file that is out of control of the plugin is also not ideal since docusaurus api doesn't specify these files will always be present for use.

Reply:
Actually I tried and it works fine with `@generated/docusaurus.config`, as we have a webpack alias

I think it's simpler to just use this value instead of the file path so I'll update and merge this

Reply:
I will ask Rohan to get u permission to push to our branches

Reply:
Thanks

didn't see you were all using the same fork :)

Normally it's a pre-checked checkbox on each PR (I think)